pkg/objstore/oss/oss.go

```diff
@@ -0,0 +1,380 @@
+package oss
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"math"
+	"math/rand"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	alioss "github.com/aliyun/aliyun-oss-go-sdk/oss"
+
+	"github.com/go-kit/kit/log"
+	"github.com/pkg/errors"
+	"github.com/thanos-io/thanos/pkg/objstore"
+	"github.com/thanos-io/thanos/pkg/runutil"
+	"gopkg.in/yaml.v2"
+)
+
+type Config struct {
+	Endpoint        string `yaml:"endpoint"`
+	Bucket          string `yaml:"bucket"`
+	AccessKeyID     string `yaml:"access_key_id"`
+	AccessKeySecret string `yaml:"access_key_secret"`
+}
+
+type Bucket struct {
+	name   string
+	logger log.Logger
+	client *alioss.Client
+	config Config
+	bucket *alioss.Bucket
+}
+
+func NewTestBucket(t testing.TB) (objstore.Bucket, func(), error) {
```

Comment:
Nope, this is the idiom for now. This is because _test.go files even exported methods are not exported between packages.

On the other hand this method is really the same across objectstores so we might want to refactor it to one (: